pkg/workloads/lib/package.py

```diff
@@ -39,12 +40,27 @@ def get_build_order(python_packages):
 
 
 def get_restricted_packages():
-    cortex_packages = {"pyspark": "2.4.0", "tensorflow": "1.12.0"}
+    req_list = ["pyspark==2.4.0", "tensorflow==1.12.0"]
     req_files = glob.glob("/src/**/requirements.txt", recursive=True)
+
     for req_file in req_files:
+        # clean requirements file, like removing comments
         with open(req_file) as f:
             for req in requirements.parse(f):
-                cortex_packages[req.name] = req.specs[0][1]
+                specifiers = [op + version for op, version in req.specs]
+                req_list.append(req.name + ",".join(specifiers))
+
+    cortex_packages = {}
+
+    for req_line in req_list:
+        parsed_req = Requirement(req_line)
+        if cortex_packages.get(parsed_req.name) is None:
+            cortex_packages[parsed_req.name] = parsed_req.specifier
+        else:
+            cortex_packages[parsed_req.name] = (
+                cortex_packages[parsed_req.name] & parsed_req.specifier
```

Comment:
what is this notation?

Reply:
Combine specifications. For example, if in spark you need numpy > 1.7 and in tensorflow you need numpy > 1.14 then the by combining the specs you have a spec that can only be satsified by numpy > 1.14